R/textplot_wordcloud.R

```diff
@@ -1,70 +1,472 @@
+#' @param title.size deprecated argument
+#' @keywords internal
+#' @author Ian Fellows
+wordcloud_comparison <- function(x, min_size, max_size, max_words,
+                                 color, font, adjust, rotation,
+                                 random_order, random_color, ordered_color,
+                                 labelcolor, labelsize, labeloffset, fixed_aspect,
+                                 # deprecated arguments
+                                 colors, scale, min.freq, max.words, 
+                                 random.order, rot.per, use.r.layout, title.size,
+                                 ...) {
+    
+    if (!missing(min.freq)) {
+        warning('min.freq is deprecated; use dfm_trim() before textplot_wordcloud()', call. = FALSE)
+        x <- dfm_trim(x, min_count = min.freq)
+    }
+    arg_dep <- character()
+    if (!missing(colors)) {
+        color <- colors
+        arg_dep <- c(arg_dep, 'color' = 'colors')
+    }
+    if (!missing(scale)) {
+        max_size <- scale[1]
+        min_size <- scale[2]
+        arg_dep <- c(arg_dep, 'min_size and max_size' = 'scale')
+    }
+    if (!missing(max.words)) {
+        max_words <- max.words
+        arg_dep <- c(arg_dep, 'max_words' = 'max.words')
+    }
+    if (!missing(random.order)) {
+        random_order <- random.order
+        arg_dep <- c(arg_dep, 'random_order' = 'random.order')
+    }
+    if (!missing(rot.per)) {
+        rotation <- rot.per
+        arg_dep <- c(arg_dep, 'rotation' = 'rot.per')
+    }
+    if (!missing(use.r.layout)) {
+        warning('use.r.layout is no longer used', call. = FALSE)
+    }
+    if (!missing(title.size)) {
+        labelsize <- title.size
+        arg_dep <- c(arg_dep, 'labelsize' = 'title.size')
+    }
+    if (length(arg_dep)) {
+        warning(paste(arg_dep), " is deprecated; use ", paste(names(arg_dep)), " instead", call. = FALSE)
+    }
+    
+    font <- check_font(font)
+    x <- x / rowSums(x)
+    x <- x - rowMeans(x)
+    x <- t(as.matrix(x))
+
+    ndoc <- ncol(x)
+    theta_bins <- seq(0, 2 * pi, length = ndoc + 1)
+
+    if (is.null(color) < ndoc)
+        color <- RColorBrewer::brewer.pal(8, "Paired")
+    group <- apply(x, 1, which.max)
+    word <- rownames(x)
+    freq <- apply(x, 1, max)
+    
+    tails <- "g|j|p|q|y"
+    nc <- length(color)
+    
+    ord <- rank(-freq, ties.method = "random")
+    word <- word[ord <= max_words]
+    freq <- freq[ord <= max_words]
+    group <- group[ord <= max_words]
+    
+    if (random_order) {
+        ord <- sample.int(length(word))
+    } else {
+        ord <- order(freq, decreasing = TRUE)
+    }
+    word <- word[ord]
+    freq <- freq[ord]
+    group <- group[ord]
+    theta_step <- 0.05
+    r_step <- 0.05
+    
+    graphics::plot.new()
+    op <- graphics::par(no.readonly = TRUE)
+    graphics::par(mar = c(0, 0, 0, 0), family = font)
+    if (labelsize > 0) {
+        graphics::plot.window(c(-0.1, 1.1), c(-0.1, 1.1), asp = 1)
```

Comment:
This is margin for comparison cloud. We can make this adjustable through `margin`.

Reply:
Better to calculate it and just get it right so that users don't have to. `wordcloud::comparison.cloud()` does this but adds so much margin that it is overkill. Can't we do something in between?

Reply:
When I clear all plots, and run this the first time
```r
textplot_wordcloud(obama_dfm, max_words = 500, rotation = 0.25, 
                   color = rev(RColorBrewer::brewer.pal(10, "RdBu")))
```
I see:
<img width="886" alt="screenshot 2018-02-23 15 38 03" src="https://user-images.githubusercontent.com/2182246/36602405-dbfa04e6-18af-11e8-8fbf-d2c13799cc59.png">

Then when I repeat the command a second time, I see:
<img width="885" alt="screenshot 2018-02-23 15 38 14" src="https://user-images.githubusercontent.com/2182246/36602362-bebfb268-18af-11e8-940f-e2b3dbd41f5f.png">

It's as if the `par(op)` command does not reset the graphics parameters.

Reply:
Or, it has something to do with verticality of the large words in the second plot.

Reply:
I now understood the issue. Pushed the fix.

Reply:
I just tested it and it still occurs, for both comparison = FALSE and for comparison = TRUE. How to verify:

1. Clear the plot history using the broom icon in the Plots window. (Same as initializing a new graphics output device.
2. Run the example plot command. This produces a plot which looks correct.
3. Repeat the same plot command. This produces a plot with the top and bottoms sliced off. It actually looked worse when I tried it now than before. Happens for both values of comparison.

Reply:
I could see edge of the cloud to be cut off when I repeat the command, but fix seems to be easy. Please try again.